### PR TITLE
[MNOE-209] Unlimited failed attempts of login

### PR DESCRIPTION
**Pull request by x4d3**
user session is not returning either not found, either a user with password_not valid if the password is valid
Resource needed to be returned properly to be able to be updated in the devise-3.5.10/lib/devise/models/lockable.rb

### CODE REVIEW

**Comment by x4d3** [comment]
@ouranos Specs are passing

**Comment by x4d3** [comment]
@ouranos I've rebased and squashed. Ready for final review.
You may test that the locking works on an express by adding in an express pointing to that version.
config/initializers/devise.rb
```Devise.setup do |config|
  config.lock_strategy = :failed_attempts
  config.maximum_attempts = 3
  config.last_attempt_warning = true
  config.paranoid = false
end
```

File: core/lib/devise/strategies/remote_authenticatable.rb [hunk 1]
```diff
@@ -1,19 +1,20 @@
...
       # Method called by warden to authenticate a resource.
       def authenticate!
         # authentication_hash doesn't include the password
         auth_params = params[scope]
-        
+
+
         # mapping.to is a wrapper over the resource model
-        resource = mapping.to.new
+        resource = mapping.to.authenticate(auth_params)
```
Comment (ouranos) [inline core/lib/devise/strategies/remote_authenticatable.rb]: I'd keep `mapping.to.remote_authenticate(auth_params)`

File: core/lib/devise/strategies/remote_authenticatable.rb [hunk 2]
```diff
@@ -24,9 +25,10 @@ def authenticate!
         #
         # If the block returns true the resource will be loged in
         # If the block returns false the authentication will fail!
-        if validate(resource){ resource = resource.remote_authentication(auth_params) }
+        if  validate(resource){ resource.password_valid }
```
Comment (ouranos) [inline core/lib/devise/strategies/remote_authenticatable.rb]: Extra space

File: core/lib/devise/strategies/remote_authenticatable.rb [hunk 3]
```diff
@@ -24,9 +25,10 @@ def authenticate!
...
           success!(resource)
         end
+
```
Comment (ouranos) [inline core/lib/devise/strategies/remote_authenticatable.rb]: Extra line

**Comment by x4d3** [comment]
@ouranos I've addressed your comments.

File: core/lib/devise/strategies/remote_authenticatable.rb (hunk 1 above)
Comment (x4d3) [reply core/lib/devise/strategies/remote_authenticatable.rb]: you meant `mapping.to.new.remote_authentication(auth_params)` ?

**Pull request merged**